src/store/modules/actions/staticData.js

```diff
@@ -38,20 +38,26 @@ export async function fetchNFTPurchaseInfo({ commit }, classId) {
 
 export async function fetchNFTMetadata({ commit }, classId) {
   let metadata;
-  const [apiMetadata = {}, chainMetadata = {}] = await Promise.all([
-    this.$api
-      .$get(api.getNFTMetadata({ classId }))
-      // eslint-disable-next-line no-console
-      .catch(err => console.error(err)),
-    getClassInfo(classId),
-  ]);
+  const chainMetadata = await getClassInfo(classId);
   const {
     name,
     description,
+    uri,
     data: { parent, metadata: classMetadata = {} } = {},
-  } = chainMetadata;
-  metadata = { name, description, metadata: classMetadata, parent };
-  if (apiMetadata) metadata = { ...metadata, ...apiMetadata };
+  } = chainMetadata || {};
+  metadata = {
+    name,
+    description,
+    metadata: classMetadata,
+    parent,
+  };
+  if (isValidHttpUrl(uri)) {
+    const apiMetadata = await this.$api
+      .$get(uri)
```

Comment:
This differ from our current behaviour (which is not ideal) of always hitting nft api, need to test if this causes any unexpected issue on prod

Reply:
I see. And I did check that [the genesis Writing NFT](https://mainnet-node.like.co/cosmos/nft/v1beta1/classes/likenft1jvywpl0f0zndhm6h8sp44esaj88mrkew226wv30ymeq5pnvux4tq760qd7) on prod has the `uri` field.